src/notifications/diskspacenotifier.cpp

```diff
@@ -63,9 +63,24 @@ void DiskSpaceNotifier::handleDiskSpaceChange(const QString &path, int percentag
         QVariantHash hints;
         hints.insert(NotificationManager::HINT_CATEGORY, "x-nemo.system.diskspace");
         //% "Getting low with storage. Please check."
-        hints.insert(NotificationManager::HINT_PREVIEW_BODY, qtTrId("qtn_memu_memlow_notification_src"));
+        QString diskLowText = qtTrId("qtn_memu_memlow_notification_src");
+        hints.insert(NotificationManager::HINT_PREVIEW_BODY, diskLowText);
         // TODO go to some relevant place when clicking the notification
         notificationId = manager->Notify(qApp->applicationName(), 0, QString(), QString(), QString(), QStringList(), hints, -1);
+
+        bool publishPermanent = true;
+        NotificationList notifications = manager->GetNotifications(qApp->applicationName());
+        foreach (LipstickNotification* notification, notifications.notifications()) {
+            if (notification->category() == "x-nemo.system.diskspace.permanent") {
+                publishPermanent = false;
+                break;
+            }
+        }
+        if (publishPermanent) {
+            QVariantHash permanentHints;
+            permanentHints.insert(NotificationManager::HINT_CATEGORY, "x-nemo.system.diskspace.permanent");
+            manager->Notify(qApp->applicationName(), 0, QString(), diskLowText, QString(), QStringList(), permanentHints, -1);
```

Comment:
Looks like this only gets removed when class is initialized, i.e. Lipstick is restarted. Would expect permanent warning to get removed as soon as space becomes available, no?

Also, is there really need for two separate notifications? Could have expected notification with all hints set being shown both as banner and as permanent item.